src/lem/eval.rs

```diff
@@ -236,24 +236,45 @@ pub fn evaluate_simple<F: LurkField, C: Coprocessor<F>>(
     evaluate_simple_with_env(func_lang, expr, store.intern_nil(), store, limit)
 }
 
+pub struct EvalConfig<'a, F: LurkField, C: Coprocessor<F>> {
+    lang: &'a Lang<F, C>,
+    ivc: bool,
+}
```

Comment:
Would the following make sense?

```rust
#[non_exhaustive]
pub enum FoldingKind {
  IVC,
  NIVC,
}

pub struct EvalConfig<F: LurkField, C: Coprocessor<F>> {
  lang: Arc<Lang<F, C>>,
  fk: FoldingKind,
}

pub enum FoldingConfig<F: LurkField, C: Coprocessor<F>> {
    ec: EvalConfig<F, C>,
    reduction_count: usize
}
```

Reply:
It does. In fact, I tried something like that in another PR (I called it `FoldingMode` instead of `FoldingKind`). But @porcuquine didn't want that kind of encoding for `FoldingConfig` and asked me to leave it as an enum

Reply:
@huitseeker I think this is the kind of thing for which we will have more data in the future. Especially if we decide to drop IVC.

This PR has gone beyond its initial scope already and the proliferation of flags is over.